### ISSUE TITLE
WSLg desktop integration does not show apps in Start menu and correct icons for GUI apps

### ISSUE THREAD

**Issue by mprasil** [nix-community/NixOS-WSL#362]
## Problem
No applications appear in Windows Start menu (like they do for Ubuntu for example) and when I launch GUI apps in NixOS WSL, the windows get generic "tux" icons in the windows taskbar.

The problem is that the WSLg layer looks up the *.desktop files in these folders:
- /usr/share/applications
- /usr/local/share/applications
- /var/lib/snapd/desktop/applications
- /var/lib/flatpak/exports/share/applications

None of which exists in NixOS by default. 

## Solution

I have created a symlink to per-user profile for the default WSL user with something like this:

```nix
systemd.tmpfiles.settings.wsl = {                                           
  "/usr/share/applications".L = {                                           
    argument = "/etc/profiles/per-user/${wsl_user.name}/share/applications";
  };                                                                        
  "/usr/share/icons".L = {                                                  
    argument = "/etc/profiles/per-user/${wsl_user.name}/share/icons";       
  };                                                                        
};                                                                          
```

In the above code, I'm also linking the icons folder for WSLg to pick up icons correctly. This makes WSLg pick up the desktop files user might have installed:

```
# relevant parts of /mnt/wslg/weston.log
...
[09:24:57.906] app_list_monitor_thread: inotify_add_watch(/usr/share/applications)
[09:24:57.906] app_list_monitor_thread: monitor /usr/share/applications
...
[09:24:57.926] desktop file: /usr/share/applications/emacs.desktop
[09:24:57.926]     Name[en_US]:Emacs (NixOS)
[09:24:57.926]     Exec:emacs
[09:24:57.926]     TryExec:(null)
[09:24:57.926]     WorkingDir:(null)
[09:24:57.926]     Icon name:emacs
[09:24:57.926]     Icon SVG :0
[09:24:57.926]     Icon file:/usr/share/icons/hicolor/128x128/apps/emacs.png
[09:24:57.926]     Icon image:0x7f39c800bdb0
[09:24:57.926] app list entry updated: Key:emacs, Name:Emacs (NixOS)
...
```

This makes the application appear in Windows Start menu:

![image](https://github.com/nix-community/NixOS-WSL/assets/8255364/a3c46704-ad23-4ff9-9434-43007a470a63)

And also makes the windows show up with correct icon in task bar:

![image](https://github.com/nix-community/NixOS-WSL/assets/8255364/77d6dd97-9240-4db1-a66b-1b1f3449f56e)

I'm not sure that is the best solution. I'm fairly new to Nix, but perhaps you could implement something similar to get a little bit better experience for GUI apps out of the box.

I've found some of the above information in [this thread](https://github.com/microsoft/wslg/issues/614).

**Comment by SuperSandro2000** [comment]
That sounds like a reasonable solution to me. Please create a PR with it :)

**Comment by mprasil** [comment]
As per discussion [here](https://github.com/nix-community/NixOS-WSL/pull/365) this was actually implemented and can be enabled by setting 

```nix
wsl.startMenuLaunchers = true;
```

However this did not work for me due to specific issue explained (and hopefully fixed) in #368
